packages/gitbook/src/components/SitePage/fetch.ts

```diff
@@ -67,16 +68,31 @@ async function resolvePage(context: GitBookSiteContext, params: PagePathParams |
         }
 
         // If a page still can't be found, we try with the API, in case we have a redirect at site level.
-        const resolvedSiteRedirect = await getDataOrNull(
-            context.dataFetcher.getSiteRedirectBySource({
-                organizationId,
-                siteId: site.id,
-                source: rawPathname.startsWith('/') ? rawPathname : `/${rawPathname}`,
-                siteShareKey: shareKey,
-            })
-        );
-        if (resolvedSiteRedirect) {
-            return redirect(resolvedSiteRedirect.target);
+        const redirectPathname = withLeadingSlash(rawPathname);
+        if (/^\/[a-zA-Z0-9-_.\/]+[a-zA-Z0-9-_.]$/.test(redirectPathname)) {
```

Comment:
Use `SITE_REDIRECT_SOURCE_PATH_PATTERN` from `@gitbook/api-client`

Reply:
This is only in gbx, it's not the same client here